### PR TITLE
[Ready] Feat/schema

### PR DESCRIPTION
**Pull request by simontegg**

### CODE REVIEW

File: src/screens/Map.tsx [hunk 1]
```diff
@@ -2,12 +2,25 @@ import * as React from 'react';
 import { View } from 'react-native';
 import { Button, Text } from 'react-native-elements';
 
-import WithHeading from '../components/WithHeading'
+import fetchLocations from '../hooks/fetch-locations'
```
Comment (simontegg) [inline src/screens/Map.tsx:6]: This is where the main stuff happens

File: scripts/generate-fake-data.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+const fs = require('fs')
+const casual = require('casual')
```
Comment (agentlewis) [inline scripts/generate-fake-data.js:2]: 👍

File: scripts/generate-fake-data.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+const visits = []
+  // id: ID!
+  // userId: ID!
+  // user: User! @connection(fields: ["userId"])
+  // locationId: ID!
+  // location: Location! @connection(fields: ["locationId"])
+  // startTime: AWSDateTime!
+  // endTime: AWSDateTime!
+for (let i = 0; i < USER_COUNT; i++) {
+  const user = users[i]
```
Comment (agentlewis) [inline scripts/generate-fake-data.js:59]: 👍

File: scripts/generate-fake-data.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+const seedfile = {
+  RequestItems: {
+    [`Location-${KEY}-${ENV}`]: locations,
+    [`User-${KEY}-${ENV}`]: users,
+    [`Visit-${KEY}-${ENV}`]: visits
+  }
+}
+
+fs.writeFileSync(path.join(__dirname, 'seeds', `seed-${ENV}.json`), JSON.stringify(seedfile, null, 2))
```
Comment (agentlewis) [inline scripts/generate-fake-data.js:86]: 👍

File: src/Navigation.tsx [hunk 5]
```diff
@@ -1,20 +1,37 @@
...
 import Dashboard from './screens/Dashboard'
 import Activity from './screens/Activity'
 import Map from './screens/Map'
 import Settings from './screens/Settings'
 import Assessment from './screens/Assessment'
 
+// Configuring here means other Amplify services "just work" when imported 
+// no need to pass down an Amplify object in a <Provider/>
+import awsconfig from './aws-exports';
+Amplify.configure(awsconfig);
```
Comment (agentlewis) [inline src/Navigation.tsx:22]: 👍

File: src/Navigation.tsx [hunk 6]
```diff
@@ -1,20 +1,37 @@
...
+
+const client = new AWSAppSyncClient({
+  url: awsconfig.aws_appsync_graphqlEndpoint,
+  region: awsconfig.aws_appsync_region,
+  auth: {
+    type: AUTH_TYPE.AMAZON_COGNITO_USER_POOLS,
+    // https://stackoverflow.com/a/55875803
+    jwtToken: async () => (await Auth.currentSession()).getAccessToken().getJwtToken()
```
Comment (agentlewis) [inline src/Navigation.tsx:30]: 👍

File: src/graphql/mutations.ts [hunk 7]
```diff
@@ -0,0 +1,232 @@
+// tslint:disable
+// eslint-disable
+// this is an auto generated file. This will be overwritten
```
Comment (agentlewis) [inline src/graphql/mutations.ts:3]: Its so nice to not have to write these!!!!

File: src/hooks/fetch-locations.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        console.log('error: ', error)
+      }
+    }
+
+    queryLocations()
+  }, [])
+
+  return [locations, loading, error, setState]
+}
+
```
Comment (agentlewis) [inline src/hooks/fetch-locations.ts:25]: Have I understood this correctly - This is just going to get all locations in the db atm?

File: amplify/scripts/amplify-push.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+const path = require('path');
+const os = require('os');
+const ini = require('ini');
+const { spawn } = require('child_process');
+const inquirer = require('inquirer');
+
+const amplify = /^win/.test(process.platform) ? 'amplify.cmd' : 'amplify';
+const dotAWSDirPath = path.normalize(path.join(os.homedir(), '.aws'));
+const configFilePath = path.join(dotAWSDirPath, 'config');
+
```
Comment (agentlewis) [inline amplify/scripts/amplify-push.js:11]: @simontegg I'm going to need a walkthrough on this file to understand it. If we jump on a call we can record it and save the video as documentation for the file.

File: amplify/backend/api/safe2/schema.graphql [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+{
+  id: ID!
+  userId: ID!
+  user: User! @connection(fields: ["userId"])
+  locationId: ID!
+  location: Location! @connection(fields: ["locationId"])
+  startTime: AWSDateTime! 
+  endTime: AWSDateTime!
+}
+
```
[the remark this answers is not in the document]
Comment (agentlewis) [reply amplify/backend/api/safe2/schema.graphql:39]: Yes absolutely.

I will sit with this and go through the prototype again.

**Pull request merged**